identify

```diff
@@ -45,4 +45,4 @@ case "$arg" in
     help
 esac
 
-ls -1 db/*.so | xargs $tool | grep -- "$regex" | perl -n -e '/db\/(.*)\.so:.*/&&print "id $1\n"'
+ls -1 db/*.so | xargs $tool | grep -- "$regex" | perl -n -e '/db\/([^ ]*)\.so/&&print "id $1\n"'
```

Comment:
I assume you added the colon to filter the `file` output (for `bid` filter) and it broke the filtering for the other methods (`sha1sum` etc). I think your solution is fine, however no need to change `(.*)` into `([^ ]*)` I think?